scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -284,6 +288,34 @@ object Execution {
   }
 
   /**
+    * This is a Thread used as a shutdown hook to clean up temporary files created by some Execution
+    */
+  case class TempFileCleanup(filesToCleanup: mutable.Set[String], mode: Mode) extends Thread {
+    val LOG = LoggerFactory.getLogger(this.getClass)
+
+    override def run(): Unit = {
+      val fs = mode match {
+        case localMode: CascadingLocal => FileSystem.getLocal(new Configuration)
+        case hdfsMode: HadoopMode =>  FileSystem.get(hdfsMode.jobConf)
+      }
+
+      filesToCleanup.foreach { file: String =>
+        try {
+          val path = new Path(file)
+          if (fs.exists(path)) {
+            // The "true" parameter here indicates that we should recursively delete everything under the given path
+            fs.delete(path, true)
+          }
+        } catch {
```

Comment:
@oscar-stripe yeah. If we for whatever reason throw an caught exception in this code that will result in the shutdown code not being run completely as I understand it.